static/js/scroll_bar.js

```diff
@@ -45,10 +45,20 @@ export function initialize() {
     const right_sidebar_width = 250;
 
     if (sbWidth > 0) {
+        $("body").addClass("has_sidebar_width");
         $(".header").css("left", "-" + sbWidth + "px");
         $(".header-main").css("left", sbWidth + "px");
         $(".header-main .column-middle").css("margin-right", 7 + sbWidth + "px");
 
+        if (user_settings.narrow_mode) {
+            $(".header-main .column-middle")[0].style.setProperty(
+                "margin-right",
+                7 + sbWidth + "px",
+                "important",
+            );
+            $(".app .app-main > .column-middle").css("right", "-" + Math.ceil(sbWidth / 2) + "px");
+        }
```

Comment:
`user_settings.narrow_mode` can change at runtime, but this only runs once on page load if it was true at load time, which can’t be right.

Reply:
Yeah, the idea is to make user reload after the setting changes. See screenshot in https://github.com/zulip/zulip/pull/19927#issuecomment-990950497